opentok-whiteboard.js

```diff
@@ -96,90 +93,95 @@ var OpenTokWhiteboard = angular.module('opentok-whiteboard', ['opentok'])
             };
             
             scope.erase = function () {
-                //scope.color = element.css("background-color") || "#fff";
-                scope.lineWidth = 50;
                 scope.erasing = true;
             };
             
             scope.capture = function () {
                 if (iOS) {
                     // On iOS you can put HTML in a mailto: link
-                    window.location.href = "mailto:?subject=Whiteboard&Body=<img src='" + canvas.toDataURL('image/png') + "'>";
+                    $window.location.href = "mailto:?subject=Whiteboard&Body=<img src='" + canvas.toDataURL('image/png') + "'>";
                 } else {
                     // We just open the image in a new window
-                    window.open(canvas.toDataURL('image/png'));
+                    $window.open(canvas.toDataURL('image/png'));
                 }
             };
 
             scope.undo = function () {
                 if (!undoStack.length)
                     return;
-                var undodata = undoStack.pop();
-                undoWhiteBoard(undodata);
-                redoStack.push(undodata);
-                sendUpdate('otWhiteboard_undo', undodata);
+                var data = undoStack[undoStack.length - 1];
+                undoWhiteBoard(data);
+                sendUpdate('otWhiteboard_undo', data);
             };
 
             var undoWhiteBoard = function (data) {
```

Comment:
Then you will probably need to go and find the corresponding part of the data to undo. As it is written right now there will be race conditions. eg. 

1. User A is drawing and they click undo.
2. User B is drawing at the same time.

When User B receives the undo message they will undo their own drawing instead of User A's drawing. So instead of just undoing whatever is last on the stack, you need to search through the data and find what User A intended to undo.